### PR TITLE
Using options "-Lkv" with curl when running the configuration checks

### PR DESCRIPTION
**Pull request by LucaCinquini**
Hi, I noticed that the configuration process fails when using your own Elasticsearch with a non-standard certificate, such as the ones issued by JPL. This PR should make the configuration process complete correctly.

### CODE REVIEW

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java [hunk 1]
```diff
@@ -1812,13 +1812,15 @@ private static int validateElasticsearch() {
 		print("checking that user provided Elasticsearch (" + elasticsearch_host + ":" + elasticsearch_port + ") is running...");
 
 		try {
-			String[] cmdArray = new String[] {"curl", "--fail", "http://" + elasticsearch_host + ":" + elasticsearch_port + "/_cluster/health"};
+			String[] cmdArray = new String[] {"curl", "-Lkv", "--fail", "http://" + elasticsearch_host + ":" + elasticsearch_port + "/_cluster/health"};
 
 			if (elasticsearch_use_auth.equalsIgnoreCase("Y")) {
 				// Add auth to curl
-				cmdArray = new String[] {"curl", "--fail", "-u", elasticsearch_username + ":" + elasticsearch_password, "https://" + elasticsearch_host + ":" + elasticsearch_port + "/_cluster/health"};
+				cmdArray = new String[] {"curl", "-Lkv", "--fail", "-u", elasticsearch_username + ":" + elasticsearch_password, "https://" + elasticsearch_host + ":" + elasticsearch_port + "/_cluster/health"};
```
Comment (ztaylor54) [inline cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:1819]: @LucaCinquini Is `-k` the only necessary option to add here? I'm not sure it's necessary to follow redirects & add verbosity to the response. 

The [docs for -k](https://curl.se/docs/manpage.html#-k) would suggest that it's the only necessary option to ignore invalid / unrecognized certs.

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java [hunk 2]
```diff
@@ -1812,13 +1812,15 @@ private static int validateElasticsearch() {
...
 			}
-
+			print("Executing command:");
+			print(String.join(" ", cmdArray));
```
Comment (ztaylor54) [inline cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:1822]: Please remove these debug print statements, as they conflict with [later output formatting](https://github.com/NASA-AMMOS/common-workflow-service/blob/5a55b3de64fd17ce9edc6f6e6fbc3b706e8e9c8d/cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java#L1830).

**Review by ztaylor54 — CHANGES_REQUESTED** [review]
Thanks for this PR, @LucaCinquini! Should be an easy merge with a few slight changes I've noted in my review.

**Comment by LucaCinquini** [comment]
Sorry for the very late action on this. I removed the print statements and only kept the "-k" option.
Please go ahead and merge this PR if possible. Thanks.

**Comment by Gillingham** [comment]
Wouldnt adding the custom CA to your systems trust store which curl uses be a better path than having curl ignore cert errors/trust issues?

**Comment by LucaCinquini** [comment]
This is actually the JPL CA, which is not part of the standard package. Yes adding the CA to each truststore is possible, but it's quite a bit of work for the user, and you need to do it in every environment you deploy CWS, and every time you upgrade the system... Quite painful.

**Comment by Gillingham** [comment]
If you don't add the CA to your system trusted CA list you need to do this sort of workaround with every piece of software on each host too. Putting the CA in a directory and running the update CA command isn't a lot of work compared with that. 

Say CWS runs python workers, or another application, you now have to figure out the correct way to add a custom CA trust to each script or language interpreter etc, or disable TLS verification like this, both those options should not be encouraged.

**Comment by LucaCinquini** [comment]
Yes you are right. Adding "-k" to the installation script is a simple change to allow the installation to proceed without too much work, when you deploy for testing. In a production environment you would definitely want to trust your CA.

**Comment by voxparcxls** [comment]
@Gillingham 
Would adding a development/testing prompt during installation be a good way to make the "-k" curl option distinct from production? How could we make this more simple?

**Comment by jamesfwood** [comment]
We have decided not to merge this PR and close it because it could compromise security in a production system.  If we want to explore this more, we should think of another way to make this work only for development environments.  Thanks!

**Pull request closed without merging**